pandas/core/frame.py

```diff
@@ -4535,12 +4535,12 @@ def _get_item(self, item: Hashable) -> Series:
     # ----------------------------------------------------------------------
     # Unsorted
 
-    def select(self, *args):
+    def select(self, *args: Hashable | list[Hashable]):
```

Comment:
So the problem with this typing is that this will accept `select(["a,"b"], ["c", "d"])` 

That's why I suggested the following:
```python
@overload
def select(self, arg0: list[Hashable] | Hashable = ...) -> pd.DataFrame: ...
@overload
def select(self, *args: Hashable) -> pd.DataFrame: ...

def select(self, arg0: list[Hashable] | Hashable = [], *args: Hashable) -> pd.DataFrame: ...
```

Reply:
Your proposal is using a default mutable parameter, which is considered a bad practice (for good reason), and I assume will also break the CI as ruff has a rule for it. I understand that the current implementation typing isn't perfect, and it could be more strict. I added a better error message if someone uses `select(["a,"b"], ["c", "d"])`, but I think this is the best we can do.

Reply:
The other way of handling that is to make the last part of the sequence of declarations:
```python
def select(self, arg0: list[Hashable] | Hashable | None = None, *args: Hashable) -> pd.DataFrame: ...
```
Then in the code if `arg0 is None`, then either `len(args)==0` (in which case it is an empty DF), or you just use `args`

Reply:
With this the typing allows `df.select(None, "col1", "col2")`, which I don't see as an improvement to the what you are trying to solve.

Reply:
> With this the typing allows df.select(None, "col1", "col2"), which I don't see as an improvement to the what you are trying to solve.

That's not correct IF you include the overloads.  If overloads are included, then only the overloads can be matched.
```python
@overload
def select(self, arg0: list[Hashable] | Hashable = ...) -> pd.DataFrame: ...
@overload
def select(self, *args: Hashable) -> pd.DataFrame: ...

def select(self, arg0: list[Hashable] | Hashable | None = None, *args: Hashable) -> pd.DataFrame: ...
```

The type checkers ONLY check the overloads, not the final declaration. So `select(None, "col1", "col2")` would be flagged by the type checker.